### ISSUE TITLE
Incompatible with other packages

### ISSUE THREAD

**Issue by Roh-codeur** [dysonance/Indicators.jl#46]
hey mate, it appears that installing Indicators downgrades HTTP. Can you please help with this

`⌅ [cd3eb016] ↓ HTTP v1.5.5 ⇒ v0.9.17`

ta!

**Comment by Roh-codeur** [comment]
it turns out the issue is with package Temporal.jl which has a hard dependency on HTTP 0.9. I have tried updating that and running tests, it all works fine. although, the URLs in that package seem out-of-date. I can submit a PR, but doesn't look like I have rights to create a branch 

is this something you can help with please?

ta!

**Comment by kpa28-git** [comment]
It would be nice if Indicators.jl didn't depend on Temporal.jl, as this prevents me from using Indicators.jl as well. The Julian way seems to be to make all packages as minimal wrt dependencies and functionality as possible, that makes for a more composable ecosystem. I think it's better to target AbstractArray and maybe Tables.jl and leave more concrete type integrations for downstream packages (ie TemporalIndicators.jl).

**Comment by smishr** [comment]
Yes if AbstractArray or Tables.jl are targetted by the package, it can be more compatible in the `DataFrames` ecosystem.

For example, I am planning to use these technical indicators on [TSFrames](https://github.com/xKDR/TSFrames.jl) objects

**Comment by smishr** [comment]
Switching to package extensions for integrating Temporal.jl could solve this issue.

JuliCon23 [demo](https://youtu.be/TiIZlQhFzyk?si=YVf86mwSzgoyKxGG), [documentation](https://pkgdocs.julialang.org/dev/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions))
